src/libs/antares/study/scenario-builder/TSnumberData.cpp

```diff
@@ -144,13 +146,16 @@ static void ApplyToMatrix(uint& errors,
                         logs.warning() << "scenario-builder: " << logprefix
                                        << "value out of bounds for the year " << (y + 1);
                 }
+                status_to_return = status_to_return && false;
```

Comment:
- `true && false == false`
- `false && false == false`

You can safely replace this line with `status_to_return = false;`

I think that we should return `false` whenever there is an error, not only if `errors > maxErrors`.

Reply:
I agree with the first part of your remark.
About the `errors > maxErrors`, this is not what I understand the code does : for me the code warns up to 20 errors (maxErrors is 20), and then stops to warn. So, at most, 20 warnings are raised.
If at least 1 error is detected, false is returned.